src/js/storage.js

```diff
@@ -244,6 +252,9 @@ BadgerPen.prototype = {
    */
   setupHeuristicAction: function(domain, action){
     this._setupDomainAction(domain, action, "heuristicAction");
+    if (this.isActionMapEmptyForDomain(domain)) {
+      this.getBadgerStorageObject('action_map').deleteItem(domain);
+    }
```

Comment:
You are correct we do not do that anywhere. It is totally preventative.

I want this layer of the API (things wrapping `_setupDomainAction`) to manage garbage collecting `action_map` objects. Without this, there is a leak in that abstraction.